packages/hmr-plugin/src/internal/hmr-lifecycle.ts

```diff
@@ -0,0 +1,61 @@
+import { NgxsBootstrapper } from '@ngxs/store/internals';
+import { Observable, Subscription } from 'rxjs';
+import { StateContext } from '@ngxs/store';
+
+import { HmrOptionBuilder } from './hmr-options-builder';
+import { HmrCallback, NgxsHmrLifeCycle } from '../symbols';
+import { HmrStateContextFactory } from './hmr-state-context-factory';
+import { HmrBeforeDestroyAction } from '../actions/hmr-before-destroy.action';
+import { HmrStorage } from './hmr-storage';
+
+export class HmrLifecycle<T extends NgxsHmrLifeCycle<S>, S> {
+  constructor(
+    private ngAppModule: T,
+    private bootstrap: NgxsBootstrapper,
+    private storage: HmrStorage<S>,
+    private context: HmrStateContextFactory<T, S>,
+    private options: HmrOptionBuilder<T, S>
+  ) {}
+
+  public hmrNgxsStoreOnInit(hmrAfterOnInit: HmrCallback<S>) {
+    if (typeof this.ngAppModule.hmrNgxsStoreOnInit === 'function') {
+      this.stateEventLoop((ctx, state) => {
+        this.ngAppModule.hmrNgxsStoreOnInit(ctx, state);
+        hmrAfterOnInit(ctx, state);
+      });
+    }
+  }
+
+  public hmrNgxsStoreBeforeOnDestroy(): Partial<S> {
+    let state: Partial<S> = {};
+    const ctx: StateContext<S> = this.context.createStateContext();
+    if (typeof this.ngAppModule.hmrNgxsStoreBeforeOnDestroy === 'function') {
+      state = this.ngAppModule.hmrNgxsStoreBeforeOnDestroy(ctx);
+    }
+
+    ctx.dispatch(new HmrBeforeDestroyAction(state));
+    return state;
+  }
+
+  private stateEventLoop(frame: HmrCallback<S>) {
+    if (this.storage.existHmrStorage) {
```

Comment:
Please change to:
```ts
if (!this.storage.existHmrStorage) {
  return;
}

...
```

Reply:
Yes, I think that a guard clause here would be cleaner.

Reply:
Done

Reply:
I unresolved this only just you won't miss my comment.
Personally, I prefer always using `{}`, even for one line statement.